module/VuFind/src/VuFind/View/Helper/Root/Citation.php

```diff
@@ -223,56 +262,71 @@ public function getDataCSL()
+                'edition' => $this->details['edition'],
+                'ISBN' => $this->driver->getISBNs(),
+                'language' => $this->driver->getLanguages(),
+                'publisher' => $this->driver->getPublishers(),
+                'publisher-place' => $this->driver->getPlacesOfPublication(),
+            ]
+        );
+
+        // journal meta
+        $this->addIfNotEmpty(
+            $item,
+            [
+                'ISSN' => $this->driver->getISSNs(),
+                'volume' => $this->driver->getContainerIssue(),
+                'volume-title' => $this->driver->getContainerVolume(),
+                // TODO: journalAbbreviation
+            ]
+        );
+        $pageFirst = $this->driver->tryMethod('getContainerStartPage');
+        $pageLast = $this->driver->tryMethod('getContainerEndPage');
+        if (!empty($pageFirst)) {
+            if (!empty($pageLast)) {
+                $item['page-first'] = $pageFirst;
+                $item['number-of-pages'] = $pageLast - $pageFirst;
+            } else {
+                $item['page'] = $pageFirst;
             }
         }
-        // pubDate -> issued
+
+        // pubDate -> issued (date)
         if (!empty($this->details['pubDate'])) {
-            $item['issued'] = ['raw' => $this->details['pubDate']];
-        }
-        // edition
-        if (!empty($this->details['edition'])) {
-            $item['edition'] = $this->details['edition'][0];
+            $item['issued'] = ['raw' => $this->getYear()];
         }
-        // isbn
-        if (!empty($this->driver->getISBNs())) {
-            $item['ISBN'] = $this->driver->getISBNs()[0];
-        }
-        // issn
-        if (!empty($this->driver->getISSNs())) {
-            $item['ISSN'] = $this->driver->getISSNs()[0];
-        }
-        // call-number
-        if (!empty($this->driver->getCallNumbers())) {
-            $item['call-number'] = $this->driver->getCallNumbers()[0];
-        }
-        // publisher
-        if (!empty($this->driver->getPublishers())) {
-            $item['publisher'] = $this->driver->getPublishers()[0];
-        }
-        // publisher-place
-        if (!empty($this->driver->getPlacesOfPublication())) {
-            $item['publisher-place'] = $this->driver->getPlacesOfPublication()[0];
+
+        // today -> accessed (date)
+        $item['accessed'] = ['raw' => date('Y-m-d\TH:i:s')];
+
+        // authors
+        if (!empty($this->details['authors'])) {
+            foreach ($this->details['authors'] as $i => $author) {
+                $item['author'][] = array_merge(
+                    ['literal' => $author],
+                    $this->nameToGivenFamily($author)
+                );
+            }
         }
         // URL
         if (!empty($this->driver->getURLs())) {
             $item['URL'] = $this->driver->getURLs()[0]['url'];
         }
-        // TODO: volume
-        // TODO: editor
-        // TODO: director
-        // TODO: journalAbbreviation
```

Comment:
Are any of these TODOs still needed?

Reply:
I moved these TODOs elsewhere in the file to be sure.

Reply:
Thanks!